orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,180 @@
+import unittest
+
+from unittest.mock import MagicMock, patch, call
+from typing import Generator, TypeVar, Optional
+from collections.abc import Iterable
+
+from hca_orchestration.contrib.argo_workflows import ExtendedArgoWorkflow, generate_argo_archived_workflows_client, \
+    ArgoFetchListOperation, ArgoArchivedWorkflowsClient
+from argo.workflows.client.models import V1alpha1Workflow, V1ObjectMeta, V1alpha1WorkflowSpec, V1alpha1Arguments, \
+    V1alpha1Parameter, \
+    V1alpha1WorkflowStatus
+
+from hca_orchestration.tests.support.mock_workflows import mock_argo_workflow
+
+
+def extend_workflow(workflow: V1alpha1Workflow) -> ExtendedArgoWorkflow:
+    return ExtendedArgoWorkflow(workflow, argo_url='https://nonexistentsite.test', access_token='token')
+
+
+class ArgoArchivedWorkflowsClientTestCase(unittest.TestCase):
+    def setUp(self):
+        self.client = ArgoArchivedWorkflowsClient("https://zombo.com", "tokentokentoken")
+
+    def workflows_to_page_of_results(self, archived_workflows: [], next_page: Optional[str]):
```

Comment:
you'll want to fix up the type annotation here - `archived_workflows` should be annotated as a list of mock v1alpha1 workflows. that's a little bit unintuitive to write out, so let me know if you need assistance there.